### PR TITLE
fix translator bug, simplify language-transformer

### PR DESCRIPTION
**Pull request by StefanVukovic99**
Currently, scanning さます brings up 去る, すます bring up する etc.
As far as I can tell, this is caused by the `_partOfSpeechToConditionFlagsMap` not being constructed correctly, which is then referenced indirectly in translator. Switching to the working `_conditionTypeToConditionFlagsMap` fixes it. This leaves `_partOfSpeechToConditionFlagsMap` with no references so I've removed it.

In LanguageTransformer, `partsOfSpeech` are a subset of `conditionTypes`, I think the intent behind handling them separately is to only allow the `partsOfSpeech` subset on dictionary entries. I'm not sure this is necessary, if it is, I think  instead of having two "bit flag trees", we could do it more simply either by adding a list like 
```json 
"partsOfSpeech": ["v1", "v5", "vk", "vs", "vz", "adj-i"]
``` 
to `LanguageTransformDescriptor` or by adding a property like `isPartOfSpeech` to `Condition`.

### CODE REVIEW

**Comment by toasted-nutbread** [comment]
I'll look into this in the next few days as well, the "v" grouping does seem to break down and match too many POSs compared to what it should.

**Comment by djahandarie** [comment]
BTW, would you happen to know if this is a regression released in a version between 24.01.14.0 (stable) and 24.2.12.0 (testing)? If yes, it would mean I should hold off on promoting testing to stable until we fix this

**Comment by Kuuuube** [comment]
> BTW, would you happen to know if this is a regression released in a version between 24.01.14.0 (stable) and 24.2.12.0 (testing)? If yes, it would mean I should hold off on promoting testing to stable until we fix this

Tested this on a few different versions with jitendex installed.

23.12.29.0 and 24.01.14.0: The first word shown is correct but lower down in the results it does still find the wrong ones.
24.2.12.0: The first word shown is the wrong one and the correct results are further down.
This pr: The first word shown is correct and the wrong results arent shown anywhere.

**Comment by toasted-nutbread** [comment]
So there are two ways to resolve this, and the issue technically stems from an incorrectly defined JSON file, but perhaps simplifying the structure may be easier in the long run to avoid such confusion.

1. Do what is done here, but go with the option of:
   > by adding a property like `isPartOfSpeech` to `Condition`.
   This could involve restoring the `_partOfSpeechToConditionFlagsMap` field so the map lookup check still only takes one condition.
2. Just delete the existing "partsOfSpeech" entries for the entries which are causing the problem, which is "v", "v1d", and "v1p".

The reason behind me setting it up this way was to slightly decouple the naming of the items internally from how the dictionaries refer to them.

* This was primarily done because "-te" and a few others are intermediate steps and don't correlate to a dictionary form, so dictionaries shouldn't be allowed to refer to them, even if they provide that name (incorrectly).
* Could also be used to allow using abbreviations (or non-abbreviations) that don't exactly match the dictionary form, in case there's ever a reason why shortening one is a good idea, e.g. shorter dictionary size. This is not currently done, but I could imagine a situation where it could be.

Whether this is useful or not is up for debate. I don't particularly care either way if this level of abstraction is kept or not, but at a minimum, I do think that the `isPartOfSpeech` is a useful discriminant, so it should be kept in some form.

**Comment by toasted-nutbread** [comment]
(to be clear, I set the precedent for incorrectly defining the JSON with my first PR for the language transformer changes, so once again I am to blame)

**Comment by StefanVukovic99** [comment]
I've gone for this first option, using `isPartOfSpeech`, but moved some stuff around in LanguageTransformer to try and keep it simple. 

Writing `Verb ... isPartOfSpeech: false` looks a bit strange to me. Maybe `isPartOfSpeech` should be something like `isDictionaryForm`. Or maybe inverted, with `isIntermediate: false`.

File: ext/js/language/language-transformer.js [hunk 1]
```diff
@@ -90,6 +96,27 @@ export class LanguageTransformer {
         return typeof conditionFlags !== 'undefined' ? conditionFlags : 0;
     }
 
+    /**
+     * @param {string} partOfSpeech
+     * @returns {number}
+     */
+    getConditionFlagsFromPartOfSpeech(partOfSpeech) {
```
Comment (toasted-nutbread) [inline ext/js/language/language-transformer.js]: We may as well remove the singular function as it's only used internally.

**Comment by toasted-nutbread** [comment]
isDictionaryForm sounds good if you want to rename it, otherwise I think everything looks fine.

**Review by toasted-nutbread — APPROVED** [review]

File: ext/data/language/japanese-transforms.json [hunk 2]
```diff
@@ -3,107 +3,107 @@
...
         "v": {
             "name": "Verb",
-            "partsOfSpeech": ["v1", "v5", "vk", "vs", "vz"],
             "i18n": [
                 {
                     "language": "ja",
                     "name": "動詞"
                 }
             ],
+            "isDictionaryForm": false,
```
Comment (djahandarie) [inline ext/data/language/japanese-transforms.json:12]: This feels a little odd to me, as "v" is a dictionary form. I guess it's just that we don't want people to set words as just "v" and want them to set it as one of the sub conditions? I feel like there must be a better way to name this

File: ext/data/language/japanese-transforms.json [hunk 3]
```diff
@@ -3,107 +3,107 @@
...
             "name": "Ichidan verb, dictionary form",
-            "partsOfSpeech": ["v1"],
             "i18n": [
                 {
                     "language": "ja",
                     "name": "一段動詞、辞書形"
                 }
-            ]
+            ],
+            "isDictionaryForm": false
```
Comment (djahandarie) [inline ext/data/language/japanese-transforms.json:34]: This is another case of it being a little odd. The name says "Ichidan verb, dictionary form", but it's marked as not a dictionary form lol

**Review by djahandarie — APPROVED** [review]
I think there is room for improvement here (regarding naming / abstractions) but since this is a bug fix let me merge it for now and we can revisit it in another PR if someone has an idea for improvements.

**Pull request merged**